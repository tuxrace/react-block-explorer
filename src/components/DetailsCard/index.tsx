import React from 'react'
import Typography from '@material-ui/core/Typography';
import CircularProgress from '@material-ui/core/CircularProgress';
import Grid from '@material-ui/core/Grid';

interface Props {
  loading: boolean;
  contract: any;
}

const DetailsCard = (props: Props) => {
    const {loading, contract} = props
    return <>
      {loading ? <Grid item><CircularProgress size={24} /></Grid>: 
                    <Grid item>
                        <Typography variant="h6">{contract?.symbol?.value}</Typography> 
                        <Typography variant="body1">Name: {contract?.name?.value}</Typography> 
                        <Typography variant="caption">{contract?.owner?.value}</Typography>
                        <Typography variant="body1">Total Tokens: {contract?.total_tokens?.value}</Typography> 
                        <Typography variant="body1">Decimals: {contract?.decimals?.value}</Typography> 
                    </Grid>}
    </>
}

export default DetailsCard
