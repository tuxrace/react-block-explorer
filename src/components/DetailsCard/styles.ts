const styles = {
    container: {
        display: 'flex'
    },
    header: {
        backgroundColor: '#131D32',
        padding: 24
    }
}

export default styles
