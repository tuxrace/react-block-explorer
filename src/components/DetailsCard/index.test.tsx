import { render } from '@testing-library/react'
import DetailsCard from './index'

describe('DetailsCard component', () => {
    test('renders DetailsCard component', async () => {
        const props = {
            loading: false,
            contract: {}
        }
        const {baseElement} = render(<DetailsCard {...props}/>);
        expect(baseElement).toMatchSnapshot()
    })
})
