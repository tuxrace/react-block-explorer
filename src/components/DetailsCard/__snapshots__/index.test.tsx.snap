// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsCard component renders DetailsCard component 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      />
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Name: 
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption"
      />
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Total Tokens: 
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Decimals: 
      </p>
    </div>
  </div>
</body>
`;

exports[`Header component renders header component 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-header-2 MuiPaper-elevation4"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        üõ°Ô∏è
         Clever Block Explorer
      </h6>
    </header>
  </div>
</body>
`;
